datafusion/core/tests/sqllogictests/test_files/pg_compat/pg_compat_null.slt

```diff
@@ -70,7 +70,7 @@ WITH HEADER ROW
 LOCATION '../../testing/data/csv/aggregate_test_100.csv'
 
 
-statement ok
+statement error DataFusion error: Error during planning: Mismatch between schema and batches
```

Comment:
this error is due to `Mismatch between`

```
batches_schema
"Schema { fields: [Field { name: "c1", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c2", data_type: Int8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c3", data_type: Int16, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c4", data_type: Int16, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c5", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c6", data_type: Int64, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c7", data_type: Int16, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c8", data_type: Int32, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c9", data_type: UInt64, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c10", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c11", data_type: Float32, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c12", data_type: Float64, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c13", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "n5", data_type: Null, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "n9", data_type: UInt64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }], metadata: {} }"
str_batches_schema
"Schema { fields: [Field { name: "c1", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c2", data_type: Int8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c3", data_type: Int16, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c4", data_type: Int16, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c5", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c6", data_type: Int64, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c7", data_type: Int16, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c8", data_type: Int32, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c9", data_type: UInt64, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c10", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c11", data_type: Float32, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c12", data_type: Float64, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c13", data_type: Utf8, nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "n5", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "n9", data_type: UInt64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }], metadata: {} }"

after type_coercion, schema has changed.

```
name: "n5", data_type: Null, nullable: true
name: "n5", data_type: Int32,  nullable: true
```